File: src/main/java/de/byjoker/myjfql/console/Console.java
package de.byjoker.myjfql.console;

import org.jetbrains.annotations.NotNull;

public interface Console {

    void log(String s);

    void logInfo(String s);

    void logWarning(String s);

    void logError(String s);

    void print(String s);

    void println(String s);

    String readPrompt();

    void bind(@NotNull Object... parameters);

    void clear();

}

File: src/main/java/de/byjoker/myjfql/console/Console.kt
package de.byjoker.myjfql.console

import org.jline.reader.Completer

interface Console {

    fun info(message: String)
    fun error(message: String)
    fun error(exception: Throwable)
    fun warn(message: String)
    fun debug(message: String)
    fun bind(completer: Completer)
    fun readPrompt(): String
    fun clear()

}
